versions.sh

```diff
@@ -3,7 +3,42 @@ set -Eeuo pipefail
 
 distsSuites=( "$@" )
 if [ "${#distsSuites[@]}" -eq 0 ]; then
-	distsSuites=( */*/ )
+	# when run without arguments, let's use "bashbrew" to get the canonical list of currently supported suites/codenames
+	bashbrew --version > /dev/null
+	if [ -z "${BASHBREW_LIBRARY:-}" ] || [ ! -s "$BASHBREW_LIBRARY/debian" ] || [ ! -s "$BASHBREW_LIBRARY/ubuntu" ]; then
+		tempDir="$(mktemp -d)"
+		trap 'rm -rf "$tempDir"' EXIT
+		wget -qO "$tempDir/debian" 'https://github.com/docker-library/official-images/raw/HEAD/library/debian'
+		wget -qO "$tempDir/ubuntu" 'https://github.com/docker-library/official-images/raw/HEAD/library/ubuntu'
+		export BASHBREW_LIBRARY="$tempDir"
+		bashbrew cat debian ubuntu > /dev/null
+	fi
+	dists="$(
+		bashbrew cat debian ubuntu --format '
+				{{- range .Entries -}}
+					{{- $.Tags "" false . | json -}}
+					{{- "\n" -}}
+				{{- end -}}
+			' | jq -r '
+				map(select(test("
+					# a few tags we explicitly want to ignore in our search for codenames
+					:(
+						experimental | rc-buggy # not a release
+						|
+						latest | .*stable | devel | rolling | testing # stable/development/rolling aliases
+						|
+						.* - .* # anything with a hyphen
```

Comment:
I thought `.*-.*` looked a little compressed and this helped it be more obvious, personally, but I don't feel strongly about it (especially with the comment out to the side which makes it pretty obvious :joy:) -- we can't do just `-` because of the `:`+`$` anchoring outside the parenthesis here (we're matching the full "tag" portion of the image references explicitly with this).